src/components/TypeComponents/SimpleSearch/Form/filters/SimpleSearchDateFilterField.js

```diff
@@ -29,7 +29,7 @@ const SimpleSearchDateFilterField = ({
   const { values } = useFormState();
   const { change } = useForm();
 
-  const isSetOrUnset = get(values, `${name}.comparator`) === 'isNull' || get(values, `${name}.comparator`) === 'isNotNull';
+  const comparatorPresent = isComparatorSpecialCase(values?.[name?.comparator]);
```

Comment:
Nearly there, if we could just change the name of this variable to `comparatorIsSpecialCase` then we'd be golden.